test/resolvers/TestDelegatableResolver.js

```diff
@@ -0,0 +1,136 @@
+const DelegatableResolver = artifacts.require('DelegatableResolver.sol')
+const { encodeName, namehash } = require('../test-utils/ens')
+const { exceptions } = require('../test-utils')
+const { expect } = require('chai')
+
+contract('DelegatableResolver', function (accounts) {
+  let node
+  let encodedname
+  let resolver
+  let signers
+  let deployer
+  let owner
+  let operator
+  let operator2
+
+  beforeEach(async () => {
+    signers = await ethers.getSigners()
+    deployer = await signers[0].getAddress()
+    owner = await signers[1].getAddress()
+    operator = await signers[2].getAddress()
+    operator2 = await signers[3].getAddress()
+    node = namehash('eth')
+    encodedname = encodeName('eth')
+    resolver = await DelegatableResolver.new(owner)
+  })
+
+  describe('supportsInterface function', async () => {
+    it('supports known interfaces', async () => {
+      assert.equal(await resolver.supportsInterface('0x3b3b57de'), true) // IAddrResolver
+      assert.equal(await resolver.supportsInterface('0xf1cb7e06'), true) // IAddressResolver
+      assert.equal(await resolver.supportsInterface('0x691f3431'), true) // INameResolver
+      assert.equal(await resolver.supportsInterface('0x2203ab56'), true) // IABIResolver
+      assert.equal(await resolver.supportsInterface('0xc8690233'), true) // IPubkeyResolver
+      assert.equal(await resolver.supportsInterface('0x59d1d43c'), true) // ITextResolver
+      assert.equal(await resolver.supportsInterface('0xbc1c58d1'), true) // IContentHashResolver
+      assert.equal(await resolver.supportsInterface('0xa8fa5682'), true) // IDNSRecordResolver
+      assert.equal(await resolver.supportsInterface('0x5c98042b'), true) // IDNSZoneResolver
+      assert.equal(await resolver.supportsInterface('0x01ffc9a7'), true) // IInterfaceResolver
+      assert.equal(await resolver.supportsInterface('0x4fbf0433'), true) // IMulticallable
+      assert.equal(await resolver.supportsInterface('0xdd48591c'), true) // IDelegatable
+    })
+
+    it('does not support a random interface', async () => {
+      assert.equal(await resolver.supportsInterface('0x3b3b57df'), false)
+    })
+  })
+
+  describe('addr', async () => {
+    it('permits setting address by owner', async () => {
+      await resolver.methods['setAddr(bytes32,address)'](node, operator, {
+        from: owner,
+      })
+      assert.equal(await resolver.methods['addr(bytes32)'](node), operator)
+    })
+
+    it('forbids setting new address by non-owners', async () => {
+      await exceptions.expectFailure(
+        resolver.methods['setAddr(bytes32,address)'](node, operator, {
+          from: operator,
+        }),
+      )
+    })
+  })
+
+  describe('authorisations', async () => {
```

Comment:
Can you add a test verifying that approval doesn't apply to names outside that scope (for example, approving a.b.c doesn't authorize d.b.c)?